### PR TITLE
[#393] Fix Power source component freezes

### LINKED ISSUE #393: Power source component freezes if adding a character 'e' in power field
* **I'm submitting a ...**
  - [x] bug report

* **What is the current status?**
In the power source component, if we try to manually change the power, the field allows us to only add numeric, no characters expect 'e'. On pressing enter, the whole app freezes and then, we have to quit it and start it again.

* Update
The text field allows us to also enter arithmetic operators, which also lead to a crash.

* **If the current behavior is a bug, please provide the steps to reproduce it**
> A screenshot of the issue may be sufficient for UI bugs

bug

* **What is the expected behavior?**
The field should not accept any letter.

* **Please tell us about your environment:**
  - OS with version number:  OSX 10.14.5

* **Would you like to work on it?**
No

### PR DESCRIPTION
**Pull request by suyashsingh234**
Fixes #393 

* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- [x] Bug fix
- [ ] Feature implementation
- [ ] Doc updates

* **What changes have you introduced?**
line 40 in src/screen/powersource/components/instrumentcluster.js

<input> method is set to type number but the property of input is to take the number as a string. We have to write extra code to parse it float even if input type is number.

* **Does this PR introduce a breaking change?**
Hopefully not. The power source component page is working fine

* **Preview / Steps to verify your work**:

### CODE REVIEW

File: src/screen/PowerSource/components/InstrumentCluster.js [hunk 1]
```diff
@@ -37,6 +37,7 @@ const Settings = props => {
   } = props;
 
   const onCheck = activeComponent => value => {
+    value=parseFloat(value);
```
Comment (CloudyPadmal) [inline src/screen/PowerSource/components/InstrumentCluster.js]:
```suggestion
    value = parseFloat(value);
```

**Comment by AakashMallik** [comment]
@suyashsingh234 Use ESLint.
Use VSCode and set up the extension for auto-formatting the code.

**Comment by AakashMallik** [comment]
@suyashsingh234 your approach should be to firstly convert the input field to text field. Then put the parse function in a try catch block. If you catch an error, return true, else false and set the value accordingly.. This PR will need more work. If you are in doubt, ping me.

File: src/screen/PowerSource/components/InstrumentCluster.js [hunk 2]
```diff
@@ -37,6 +37,7 @@ const Settings = props => {
   } = props;
 
   const onCheck = activeComponent => value => {
+    value = parseFloat(value);
```
Comment (AakashMallik) [inline src/screen/PowerSource/components/InstrumentCluster.js]: Need better approach

Comment (suyashsingh234) [reply src/screen/PowerSource/components/InstrumentCluster.js]: @AakashMallik  it fixes all the problem with input.

**Pull request closed without merging**